src/azure-cli/azure/cli/command_modules/vm/_actions.py

```diff
@@ -55,34 +57,58 @@ def load_images_thru_services(cli_ctx, publisher, offer, sku, location):
     def _load_images_from_publisher(publisher):
         from azure.core.exceptions import ResourceNotFoundError
         try:
-            offers = client.virtual_machine_images.list_offers(location, publisher)
+            if edge_zone is not None:
+                offers = edge_zone_client.list_offers(location, edge_zone, publisher)
+            else:
+                offers = client.virtual_machine_images.list_offers(location, publisher)
         except ResourceNotFoundError as e:
             logger.warning(str(e))
             return
         if offer:
             offers = [o for o in offers if _matched(offer, o.name)]
         for o in offers:
             try:
-                skus = client.virtual_machine_images.list_skus(location, publisher, o.name)
+                if edge_zone is not None:
+                    skus = edge_zone_client.list_skus(location, edge_zone, publisher, o.name)
+                else:
+                    skus = client.virtual_machine_images.list_skus(location, publisher, o.name)
             except ResourceNotFoundError as e:
                 logger.warning(str(e))
                 continue
             if sku:
                 skus = [s for s in skus if _matched(sku, s.name)]
             for s in skus:
                 try:
-                    images = client.virtual_machine_images.list(location, publisher, o.name, s.name)
+                    if edge_zone is not None:
+                        images = edge_zone_client.list(location, edge_zone, publisher, o.name, s.name)
+                    else:
+                        images = client.virtual_machine_images.list(location, publisher, o.name, s.name)
                 except ResourceNotFoundError as e:
                     logger.warning(str(e))
                     continue
                 for i in images:
-                    all_images.append({
-                        'publisher': publisher,
-                        'offer': o.name,
-                        'sku': s.name,
-                        'version': i.name})
-
-    publishers = client.virtual_machine_images.list_publishers(location)
+                    if edge_zone is not None:
+                        all_images.append({
+                            'publisher': publisher,
+                            'offer': o.name,
+                            'sku': s.name,
+                            'edge_zone': edge_zone,
+                            'version': i.name})
+                    else:
+                        all_images.append({
+                            'publisher': publisher,
+                            'offer': o.name,
+                            'sku': s.name,
+                            'version': i.name})
```

Comment:
How about we make this code a little simpler?
```suggestion
                    image_info = {
                            'publisher': publisher,
                            'offer': o.name,
                            'sku': s.name,
                            'version': i.name
                     }
                     if edge_zone is not None:
                            image_info['edge_zone'] = edge_zone
                     all_images.append(image_info)
                      
```